gato/policy/gato_policy.py

```diff
@@ -458,6 +467,88 @@ def predict_text(self, batch_dict, max_length=20, deterministic=True):
         return concat_logits, predicted_tokens
 
 
+        return concat_probs, concat_pred_tokens
+
+    # This funciton can be used to generate a response from an image, such as generating the caption or 
+    # an answer to a question about an image, it is adapted from the original predict_caption() function
+    def predict_response(self, image, prompt_tokens = [], max_length=128, deterministic=True):
+        """
+        image is in the format of 1 x 3 x H x W, where 1 is the num_images, 3 is the 3 RGB channels, default value for H and W is 256
+        prompt_tokens is a list of text tokens:
+            if the predicted response is the caption for the image, then it is an empty list
+            if the predicted response is an answer to a question about the image, then this list are the tokens of the question
+        max_length is the max length of response to be generated, will cut off at max_length
+        """
+        image_embeddings = self.image_embedding(image.to(self.device)) # the image embedding that will be used to generate response
+        n_images = image_embeddings.shape[0]
+        n_patches = image_embeddings.shape[1] 
+        assert n_images == 1, "number of images should always be 1 for predicting response"
+
+        response_tokens = []
+
+        for idx in range(max_length): # idx can be viewed as the index of the 'next_token' within the list of such generated response tokens
+
+            # Include 'image_embeddings' instead of 'images' below to avoid re-calculating the same image embedding in every loop
+            # Note: the following 'text' is actually text token IDs instead of actual text. 
+            # We were using actual text before (and at that time tokenize_input_dicts(...) function also took actual text), and found an error 
+            # due to an edge case. The edge case is that the generated tokens could contain a sequence of .... (periods) or ,,,, (commas), 
+            # and if actual text is used in the following 'text', the tokenizer will treat the sequence as one token (which is not the desired behavior) 
```

Comment:
The comments related to `text` can be removed. It wouldn't be helpful to someone in future.